pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -161,26 +168,15 @@ func (n Node) isPodCompatible(pod *v1.Pod, it cloudprovider.InstanceType) bool {
 // hasCompatibleResources tests if a given node selector and resource request list is compatible with an instance type
 func (n Node) hasCompatibleResources(resourceList v1.ResourceList, it cloudprovider.InstanceType) bool {
 	for name, quantity := range resourceList {
-		switch name {
-		case resources.NvidiaGPU:
-			if it.NvidiaGPUs().Cmp(quantity) < 0 {
-				return false
-			}
-		case resources.AWSNeuron:
-			if it.AWSNeurons().Cmp(quantity) < 0 {
-				return false
-			}
-		case resources.AMDGPU:
-			if it.AMDGPUs().Cmp(quantity) < 0 {
-				return false
-			}
-		case resources.AWSPodENI:
-			if it.AWSPodENI().Cmp(quantity) < 0 {
-				return false
-			}
-		default:
+		// we don't care if the pod is requesting zero quantity of some resource
+		if quantity.IsZero() {
```

Comment:
Is this a real case we need to check? Does the API Server even allow this?

Reply:
You're right, I tested and the API server just drops the zero quantity resource.

Reply:
Spoke too soon, API server does pass along an empty resource request.  I was running the wrong test. Maybe we still don't care though, I'm fine leaving or removing the check.